langserver/config.go

```diff
@@ -36,27 +46,25 @@ func (s *server) DidChangeConfiguration(ctx context.Context, params *protocol.Di
 			Message: fmt.Sprintf("Received notification change: %v\n", params),
 		})
 
-	setting := params.Settings
+	config := localLSPConfiguration{}
 
-	// the struct expected is the following
-	// promql:
-	//   url: http://
-	//   interval: 3w
-	m, ok := setting.(map[string]map[string]string)
-	if !ok {
+	// In order to avoid a loop other different interface coming from the general settings
```

Comment:
This comment is a bit confusing and has a couple of language issues. Maybe something like

```
Go doesn't provide an easy way to cast nested maps into structs.

As a workaround, the configuration is first converted back to the JSON that was provided by the client and then marshalled back into the expected structure of the settings.

If you are reading this code and are aware of a better solution to do this, feel free to submit a PR.
```

Note that, while I don't like it, I didn't find a better way to do this either.

Reply:
oh don't worry I don't like it too lol